File: 990-verifying-an-alien-dictionary/verifying-an-alien-dictionary.go
func isAlienSorted(words []string, order string) bool {
    
    ranks := make([]int, 26)
    for i, char := range order {
        ranks[char - 'a'] = i
    }

    for i := 0; i < len(words)-1; i++ {
        w1 := words[i]
        w2 := words[i+1]
        
        isSorted := false

        for k := 0; k < len(w1) && k < len(w2); k++ {
            rank1 := ranks[w1[k] - 'a']
            rank2 := ranks[w2[k] - 'a']

            if rank1 < rank2 {
                isSorted = true
                break 
            }

            if rank1 > rank2 {
                return false
            }
        }

        if !isSorted {
            if len(w1) > len(w2) {
                return false
            }
        }
    }

    return true
}

File: 990-verifying-an-alien-dictionary/verifying-an-alien-dictionary.py
class Solution:
    def isAlienSorted(self, words: List[str], order: str) -> bool:
        order_map = {char: i for i, char in enumerate(order)}
        
        for i in range(len(words) - 1):
            word1 = words[i]
            word2 = words[i + 1]
            
            for j in range(min(len(word1), len(word2))):
                char1, char2 = word1[j], word2[j]
                
                if char1 != char2:
                    if order_map[char1] > order_map[char2]:
                        return False
                    break
            else:
                if len(word1) > len(word2):
                    return False
                    
        return True